stackslib/src/chainstate/burn/db/sortdb.rs

```diff
@@ -1842,36 +1861,23 @@ impl<'a> SortitionHandleConn<'a> {
         SortitionHandleConn::open_reader(connection, &sn.sortition_id)
     }
 
-    /// Does the sortition db expect to receive unknown blocks from
-    /// this tenure?
-    ///
-    /// This is used by nakamoto nodes while they are at or near the
-    /// current chain tip: only recent tenures can receive blocks this
-    /// way. Otherwise, the `BlockHeaderHash` must have been
-    /// explicitly confirmed by a block commit.
-    pub fn expects_blocks_from_tenure(
-        &self,
-        miner_pk: &Secp256k1PublicKey,
-    ) -> Result<Option<BlockSnapshot>, db_error> {
-        let to_check = Hash160::from_node_public_key(miner_pk);
-        let mut cur_tip = self.context.chain_tip.clone();
-        for _ in 0..NAKAMOTO_TENURE_BLOCK_ACCEPTANCE_PERIOD {
-            let cur_snapshot = SortitionDB::get_block_snapshot(self.sqlite(), &cur_tip)?
-                .ok_or_else(|| db_error::NotFoundError)?;
-            if cur_snapshot.miner_pk_hash == Some(to_check) {
-                return Ok(Some(cur_snapshot));
-            }
-            cur_tip = cur_snapshot.parent_sortition_id.clone();
-        }
-        Ok(None)
-    }
-
     /// Does the sortition db expect to receive blocks from
-    /// signed by this stacker set?
+    /// signed by stacker set?
     pub fn expects_stacker_signature(
         &self,
+        consensus_hash: &ConsensusHash,
         _stacker_signature: &MessageSignature,
     ) -> Result<bool, db_error> {
+        // is this consensus hash in this fork?
+        let Some(bhh) = SortitionDB::get_burnchain_header_hash_by_consensus(self, consensus_hash)?
+        else {
+            return Ok(false);
+        };
+        let Some(_sortition_id) = self.get_sortition_id_for_bhh(&bhh)? else {
+            return Ok(false);
+        };
+
+        // TODO: query set of stacker signers in order to get the aggregate public key
         Ok(true)
```

Comment:
Yes -- the TODO ensures that the only way this method returns `true` is if the signature came from the _current_ stackers, identified by the `consensus_hash` (which in turn identifies the reward cycle its MARF'ed PoX info, which would include the aggregate public key).

Reply:
I think it cannot use the `consensus_hash` argument, but instead something like a new `tip: &ConsensusHash` argument. For example, if a node's current view is like Bitcoin block `10,000`, it should _never_ use this method for accepting blocks from Bitcoin block `6,000`, because then it is a DoS vector: a prior stacking set could sign off on blocks and broadcast them and the node would have no way to reject them. For "old" blocks, the node must use the block hashes committed to during leader block commits. This method must only accept current stacker set signatures (_not_ "current at the time the block claims to have been produced"), otherwise the trust relationship for nodes and stackers is that nodes trust prior stacker sets _indefinitely_ to not DoS them.

Reply:
I've updated this method to require that `consensus_hash` is within one reward cycle of the `SortitionHandleConn`'s tip.  During initial block download, it's possible that there will still be some blocks from the previous reward cycle in-flight while the next reward cycle is processed, so the limit must be permitted to straddle the reward cycle boundary.